src/main/java/net/jodah/typetools/TypeResolver.java

```diff
@@ -310,31 +309,24 @@ private static void populateLambdaArgs(Class<?> functionalInterface, final Class
             Method objectMethod = OBJECT_METHODS.get(m.getName());
             if (objectMethod != null && Arrays.equals(m.getTypeParameters(), objectMethod.getTypeParameters()))
               continue;
-
-            // Get functional interface's type params
-            Type returnTypeVar = m.getGenericReturnType();
-            Type[] paramTypeVars = m.getGenericParameterTypes();
             
             // Get lambda's type arguments
             ConstantPool constantPool = (ConstantPool) GET_CONSTANT_POOL.invoke(lambdaType);
-            String[] methodRefInfo = constantPool
-                .getMemberRefInfoAt(constantPool.getSize() - resolveMethodRefOffset(constantPool));
-
-            // Skip auto boxing methods
-            if (methodRefInfo[1].equals("valueOf") && constantPool.getSize() > 22) {
```

Comment:
My understanding here is that we want to skip auto-boxing methods (`valueOf`) if there is another method which seems to be a better fix, but _do_ want to return the information from the `valueOf` method in case there's no better alternative.

I believe that the old code did this somewhat implicitly by ignoring exceptions thrown in the body of this `if` case. In this patch, I decided to try to make this more explicit by moving this logic into the `getMethodRefInfo` below.